tests/MessagesTest.php

```diff
@@ -0,0 +1,75 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Yiisoft\Validator\Tests;
+
+use PHPUnit\Framework\TestCase;
+use Yiisoft\Translator\CategorySource;
+use Yiisoft\Translator\Message\Php\MessageSource;
+use Yiisoft\Translator\SimpleMessageFormatter;
+use Yiisoft\Translator\Translator;
+use Yiisoft\Validator\Rule\Regex;
+use Yiisoft\Validator\Validator;
+
+use function dirname;
+
+final class MessagesTest extends TestCase
+{
+    public function testBase(): void
+    {
+        $locale = 'ru';
+
+        $categorySource = new CategorySource(
+            Validator::DEFAULT_TRANSLATION_CATEGORY,
+            new MessageSource($this->getMessagesPath()),
+            new SimpleMessageFormatter(),
+        );
+
+        $translator = (new Translator($locale))->withLocale($locale);
+        $translator->addCategorySources($categorySource);
+
+        $validator = new Validator(translator: $translator);
+
+        $result = $validator->validate(
+            'hello',
+            new Regex('~\d+~')
+        );
+
+        $this->assertSame(
+            ['' => ['Значение неверно.']],
+            $result->getErrorMessagesIndexedByAttribute(),
+        );
+    }
+
+    public function dataNonEmpty(): array
+    {
+        return [
+            ['ru'],
+        ];
+    }
+
+    /**
+     * @dataProvider dataNonEmpty
+     */
+    public function testNonEmpty(string $locale): void
+    {
+        $file = $this->getMessagesPath() . '/' . $locale . '/' . Validator::DEFAULT_TRANSLATION_CATEGORY . '.php';
+        $this->assertFileExists($file);
+
+        $messages = require $file;
+        $this->assertIsArray($messages);
+
+        foreach ($messages as $id => $message) {
+            $this->assertIsString($id);
+            $this->assertNotEmpty($id);
+            $this->assertIsString($message);
+            $this->assertNotEmpty($message);
```

Comment:
A nitpick obviously, but since we have an agreement that every error sentence must end with the dot. we can additionally check that as well.

Reply:
There may be exceptions.

Reply:
Could you provide an example? This is not critical in any way, but I see no problem with adjusting it when the issue will arise.

Reply:
Not critical and can't be strict because of possible nuances in other languages. Closing.